tests/src/App/DocumentObject.cpp

```diff
@@ -0,0 +1,157 @@
+    // A vector of DocumentObjects used to store the result of the call to
+    // DocumentObject::getSubObjectList() with the parameters subname and subsizes
+    auto docSubObjsWithSubSizes {std::vector<DocumentObject*>()};
+    // A vector of DocumentObjects used to store the result of the call to
+    // DocumentObject::getSubObjectList() with the flatten parameter set to true
+    auto docSubObjsFlatten {std::vector<DocumentObject*>()};
+
+    // Act
+    docSubObjsNoSubName = part->getSubObjectList(nullptr);
+    docSubObjsWithSubName = part->getSubObjectList(subName.c_str());
+    docSubObjsWithSubSizes = part->getSubObjectList(subName.c_str(), &sizesNoFlatten);
+    docSubObjsFlatten = part->getSubObjectList(subName.c_str(), &sizesFlatten, true);
+
+    // Assert
+
+    // If DocumentObject::getSubObjectList() is called without giving the subname argument it
+    // returns a vector with only one entry, corresponding to the object that called the method
+    EXPECT_EQ(docSubObjsNoSubName.size(), 1);
+    EXPECT_EQ(docSubObjsNoSubName[0]->getID(), _doc->getObject(partName)->getID());
+
+    // If DocumentObject::getSubObjectList() is called with the subname argument it returns a vector
+    // with one entry for each sub object in the subname plus one entry with the object that called
+    // the method
+    EXPECT_EQ(docSubObjsWithSubName.size(), 3);
+    EXPECT_EQ(docSubObjsWithSubName[0]->getID(), _doc->getObject(partName)->getID());
+    EXPECT_EQ(docSubObjsWithSubName[1]->getID(), _doc->getObject(fuseName)->getID());
+    EXPECT_EQ(docSubObjsWithSubName[2]->getID(), _doc->getObject(boxName)->getID());
+
+    // If DocumentObject::getSubObjectList() is called with the subsizes argument it returns the
+    // same vector of the previous case and the subsizes argument stores the positions in the
+    // subname string corresponding to start of the sub objects names
+    EXPECT_EQ(docSubObjsWithSubSizes.size(), 3);
+    EXPECT_EQ(docSubObjsWithSubSizes[0]->getID(), _doc->getObject(partName)->getID());
+    EXPECT_EQ(docSubObjsWithSubSizes[1]->getID(), _doc->getObject(fuseName)->getID());
+    EXPECT_EQ(docSubObjsWithSubSizes[2]->getID(), _doc->getObject(boxName)->getID());
+    EXPECT_EQ(sizesNoFlatten.size(), 3);
+    EXPECT_EQ(sizesNoFlatten[0], 0);
+    EXPECT_EQ(sizesNoFlatten[1], 7);
+    EXPECT_EQ(sizesNoFlatten[2], 17);
+
+    // If DocumentObject::getSubObjectList() is called with the flatten argument set to true it
+    // returns a vector with all the sub objects in the subname that don't belong to the same
+    // App::GeoFeatureGroupExtension object, plus one entry with the object that called the method
+    EXPECT_EQ(docSubObjsFlatten.size(), 2);
+    EXPECT_EQ(docSubObjsFlatten[0]->getID(), _doc->getObject(partName)->getID());
+    EXPECT_EQ(docSubObjsFlatten[1]->getID(), _doc->getObject(boxName)->getID());
+    EXPECT_EQ(sizesFlatten.size(), 2);
+    EXPECT_EQ(sizesFlatten[0], 0);
+    EXPECT_EQ(sizesFlatten[1], 17);
```

Comment:
Also here, I would prefer to have the positions expressed in terms of the length of the `partName` and `boxName`.

Reply:
Done!